src/method/query/response.rs

```diff
@@ -0,0 +1,231 @@
+use std::ops::Deref;
+use std::ops::DerefMut;
+use std::ops::Index;
+use std::slice::SliceIndex;
+
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+use crate::Result;
+use crate::Value;
+
+/// A wrapper type around the list of results for the queries that were returned
+/// by the database.
+///
+/// Provides utility functions to access the result of one specific query, or if
+/// needed, all queries at once.
+///
+#[derive(Debug, Clone)]
+pub struct QueryResponse(Vec<QueryResult>);
+
+impl QueryResponse {
+    /// Constructs an empty [`QueryResponse`]
+    pub fn new() -> Self {
+        Self(Default::default())
+    }
+
+    /// Returns a reference the result for the `n`-th query from the response. If
+    /// no result is found at this index then [None] is returned.
+    pub fn get_query(&self, n: usize) -> Option<&QueryResult> {
+        self.0.get(n)
+    }
+
+    /// Returns a reference to the result for the `n`-th query from the response.
+    ///
+    /// # Panics
+    /// Unlike the [`Self::get()`] function no bounding check is performed and may panic if
+    /// `n` is out of bounds.
+    pub fn get_query_unchecked(&self, n: usize) -> &QueryResult {
+        &self.0[n]
+    }
+
+    /// Returns a reference to all of the results from the response.
+    pub fn all(&self) -> &Vec<QueryResult> {
```

Comment:
How about implementing `AsRef` instead? In future we might add an implementation for `AsMut` too if need be. We may also need `into_inner` for those who may want an owned `Vec<QueryResult>`.

Reply:
An excellent idea! I replaced that function with the `impl AsRef<Vec<QueryResult>>`, and added an `into_inner` function to the `QueryResponse` type.